### ISSUE TITLE
attempting to get a weekly-incidence example working

### ISSUE THREAD

**Issue by bbolker** [canmod/macpan2#172]
I added a file [here](https://github.com/canmod/macpan2/blob/bc-tour/misc/experiments/weekly_incidence.R) (I guess maybe it should be in its own subdirectory?) that attempts to compute and report weekly incidence, via `group_sums` (eventually it would be nice to have an example that also shows how to do this by (1) creating an accumulator compartment and differencing it in the `after` phase, (2) multiplying by a fixed matrix in the `after` phase; `group_sums()` may be the best approach for this use case but flexibility is good)

I ran into two problems that I don't know how to solve in the current framework.

- I get a "group indexes are out of range" error; I've tried this with various permutations of 1- and 0-indexing. I could certainly be making some kind of thinko but so far can't figure it out.
- I don't know how to add a matrix-to-be-reported; when I add daily incidence to the expression list (equal to recovery in this case), it doesn't show up in the report ...

**Comment by stevencarlislewalker** [comment]
Thanks Ben!  The answer to your first question is that there is a documentation bug that has been fixed and will soon be pushed (I'm traveling right now though so not until Tuesday probably.  The third argument is actually a column vector the same size as the output ... not the number of elements in the output.

The answer to the second question is to use the outputs argument in the mp_simulator function.  The 'saving' from before just refers to saving the values at each iteration so that they can be referred to in the after phase, in future iterations using lag functions or convolution, or in the objective function.

**Comment by bbolker** [comment]
Thanks. I think I have this working, although I have to admit that the change in design is a bit confusing; what is the intention of making the third argument a vector rather than a scalar denoting the length?

**Comment by stevencarlislewalker** [comment]
I don't feel strongly about it either way. But here was my thinking.

Consider the case when you have several vectors all the same size representing the same set of entities.  For example, the state, inflow, and outflow vectors are all the same size and each element represents a state.  So when using group_sums to compute inflow and outflow, you have the state vector sitting there with all of the information required to define the size of those other vectors.  Couple this with the fact that there is no length function in the engine (and that engine functions cannot _return_ integers of any kind), I felt that it was more natural to have a prototype vector as the third argument rather than requiring explicitly defining an integer for the size of the output vector.

But if it is confusing I don't mind changing it back.

Another idea, which I don't love ... I think, is if the 3rd argument is an integer (ie passed through the integers argument of mp_tmb_model_spec), then this is taken to be the size of the output.  But if the 3rd argument is a column vector matrix (passed through the defaults argument), then this is taken to be a prototype for the output.

Anyways ... very happy to discuss more.  But currently I feel like i have little information on what would be more or less confusing in a world where the documentation is stable.

**Comment by bbolker** [comment]
I think I get it; I'm not convinced, but seems fine to leave it at least for now/until after further discussion.
